spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/AbstractReactiveHealthIndicator.java

```diff
@@ -75,19 +76,24 @@ protected AbstractReactiveHealthIndicator(Function<Throwable, String> healthChec
 
 	@Override
 	public final Mono<Health> health() {
+		Mono<Health> result;
 		try {
-			return doHealthCheck(new Health.Builder()).onErrorResume(this::handleFailure);
+			result = doHealthCheck(new Health.Builder()).onErrorResume(this::handleFailure);
```

Comment:
why not return in line ?

Reply:
Take a look at line 86. `result` is used in a chain with `.doOnNext`.